pkg/manager/client/discovery.go

```
@@ -30,26 +31,46 @@ func NewDiscoveryClient(ctx context.Context, config *rest.Config) (*DiscoveryCli
 	}, nil
 }
 
-func toObj(b []byte) interface{} {
-	var objMap map[string]interface{}
+func toObj(b []byte, groupVersion string, kind string) interface{} {
```

If the function can fail, we should return the error code here as well.